File: src/simulation/culture/group/ResourceBehaviour.java
package simulation.culture.group;

import kotlin.collections.ArraysKt;
import simulation.Controller;
import simulation.space.resource.ResourcePack;

public class ResourceBehaviour {
    private PlacementStrategy placementStrategy;

    ResourceBehaviour(PlacementStrategy placementStrategy) {
        this.placementStrategy = placementStrategy;
    }

    public void procedeResources(ResourcePack resourcePack) {
        placementStrategy.place(resourcePack);
    }

    @Override
    public String toString() {
        return placementStrategy.toString();
    }

    public static ResourceBehaviour getRandom(Group group) {
        return new ResourceBehaviour(new PlacementStrategy(group.getOverallTerritory(),
                shmp.random.RandomCollectionsKt.randomElement(
                        ArraysKt.toList(PlacementStrategy.Strategy.values()),
                        Controller.session.random)
        ));
    }
}

File: src/simulation/culture/group/resource_behaviour/ResourceBehaviour.kt
package simulation.culture.group.resource_behaviour

import shmp.random.randomElement
import simulation.culture.group.centers.Group
import simulation.space.resource.MutableResourcePack
import kotlin.random.Random

class ResourceBehaviour constructor(private val placementStrategy: PlacementStrategy) {
    fun proceedResources(resourcePack: MutableResourcePack) {
        placementStrategy.place(resourcePack)
    }

    override fun toString() = placementStrategy.toString()
}

fun getRandom(group: Group, random: Random): ResourceBehaviour = ResourceBehaviour(PlacementStrategy(
        group.territoryCenter.territory,
        randomElement(
                PlacementStrategy.Strategy.values().toList(),
                random
        )
))
